lgc/util/llpcGfxRegHelper.cpp

```diff
@@ -0,0 +1,311 @@
+{
+    { 0, 30, 2 }, // FilterMode
+    { 2, 20, 2 }, // XyMagFilter
+    { 2, 22, 2 }, // XyMinFilter
+};
+
+// =====================================================================================================================
+// Helper class for handling Registers defined in SQ_IMG_SAMP_WORD
+SqImgSampRegHelper::SqImgSampRegHelper(
+    Builder*      pBuilder,      // [in] Bound builder context
+    Value*        pRegister,     // [in] Bound register vec <n x i32>
+    GfxIpVersion* pGfxIpVersion) // [in] Target gfxIp version
+: GfxRegHelper(pBuilder, pRegister)
+{
+    m_pGfxIpVersion = pGfxIpVersion;
+
+    switch (pGfxIpVersion->major)
+    {
+    case 9:
+    case 10:
+        m_bitsInfo = g_sqImgSampRegBitsGfx9;
+        break;
+    default:
+        llvm_unreachable("GfxIp is not supported!");
+        break;
+    }
+    m_bitsState = std::make_unique<BitsState[]>(static_cast<uint32_t>(SqSampRegs::SqSampRegsCount));
+}
+
+// =====================================================================================================================
+// Acquire register value
+Value* SqImgSampRegHelper::GetReg(
+    SqSampRegs id) // Register ID
+{
+    switch (id)
+    {
+    case SqSampRegs::FilterMode:
+    case SqSampRegs::xyMagFilter:
+    case SqSampRegs::xyMinFilter:
+        return GetRegCommon(static_cast<uint32_t>(id));
+    default:
+        llvm_unreachable("Not implemented!");
+        break;
+    }
+    return nullptr;
+}
+
+// =====================================================================================================================
+// Set register value
+void SqImgSampRegHelper::SetReg(
+    SqSampRegs id,     // Register ID
+    Value*     pParam) // [in] Param to be set
+{
+    switch (id)
+    {
+    case SqSampRegs::FilterMode:
+    case SqSampRegs::xyMagFilter:
+    case SqSampRegs::xyMinFilter:
+        SetRegCommon(static_cast<uint32_t>(id), pParam);
+        break;
+    default:
+        llvm_unreachable("Set \"IsTileOpt\" is not allowed!");
+        break;
+    }
+}
+
+// =====================================================================================================================
+// SqImgSampReg Bits infomation look up table (Gfx9)
+// Refer to imported/chip/gfx9/gfx9_plus_merged_registers.h : SQ_IMG_RSRC_WORD
+static constexpr BitsInfo g_sqImgRsrcRegBitsGfx9[static_cast<uint32_t>(SqRsrcRegs::SqRsrcRegsCount)] =
+{
+    { 0,  0, 32 }, // BaseAddress
+    { 1,  0,  8 }, // BaseAddressHi
+    { 1, 20,  9 }, // Format
+    { 2,  0, 14 }, // Width
+    { 2, 14, 14 }, // Height
+    { 3,  0, 12 }, // DstSelXYZW
+    { 3, 20,  5 }, // IsTileOpt
+    { 4,  0, 13 }, // Depth
+    { 4, 13, 12 }, // Pitch
+    { 4, 29,  3 }, // BcSwizzle
+    {},            // WidthLo
+    {},            // WidthHi
+};
+
+// =====================================================================================================================
+// SqImgSampReg Bits infomation look up table (Gfx10)
+// Refer to: (Not released)
```

Comment:
Can you add a `TODO` or `FIXME` to fill it in later?

Reply:
Added TODO